### ISSUE TITLE
syntax error for single-kernel broadcast over splatted arguments ("multiple splatted args cannot be fused into a single broadcast")

### ISSUE THREAD

**Issue by jrevels** [JuliaLang/julia#26127]
```julia
julia> f(args...) = *(args...)
f (generic function with 1 method)

julia> x, y, z = (1,2), 3, (4, 5)
((1, 2), 3, (4, 5))

julia> f.(x..., y, z...)
ERROR: syntax: multiple splatted args cannot be fused into a single broadcast

julia> broadcast(f, x..., y, z...)
120
```

Seems like `f.(...)` here should just lower straight to `broadcast(f, ...)`, since it's a straightforward syntactic transformation.

Is this an edge case that's just not handled correctly by the syntactic fusion code, or is the error intentional (perhaps there's an ambiguity here that I'm just missing)? At the very least, the error message should be changed - there's only a single kernel present, so there's no potential for fusion in the first place.

EDIT: edited example to avoid the syntactic fuser closing over the numeric literal (as mentioned by @mbauman)

**Comment by mbauman** [comment]
Note that it is technically "fusing" the numeric literal into `(as, bs)->f(as..., 2, bs...)`.

I do think it might be possible to handle multiple splats… it's probably just annoying and hard.

**Comment by vtjnash** [comment]
Will be fixed by #25377 (when it is merged).
